docs/features/software-catalog/extending-the-model.md

```diff
@@ -22,8 +22,59 @@ the years, and now includes ML models, Apps, data pipelines and many more.
 
 ## Adding a new type
 
-TODO: Describe what changes are needed to add a new type that shows up in the
-catalog.
+To add a new type to the catalog, you need to:
+
+1. Modify the tabs in the component page
+2. Set the new type in your component (e.g., modify your `catalog-info.yaml`)
+
+You can also optionally modify the page to display different information when
+the component is displayed (see
+[`EntityPage.tsx`](https://github.com/spotify/backstage/blob/master/packages/app/src/components/catalog/EntityPage.tsx)),
+but this example will use the `DefaultEntityPage` with the Overview and Docs
+tabs.
+
+For example, to add a new type of iOS Application, first modify the `const tabs`
+variable in `plugins/catalog/src/components/CatalogPage/CatalogPage.tsx` like
```

Comment:
As mentioned, this isn't a file that you have access to as a user of Backstage. Making this configurable is what #2791 is about